### PR TITLE
Make better

### PR DESCRIPTION
**Pull request by Fiamma-Jikukan**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -93,34 +88,32 @@ app.post("/signup", async (req, res) => {
 })
 
 app.post("/login", async (req, res) => {
-    const loginuser = await User.findOne({ username: req.body.username })
     try {
-        if (loginuser) {
+        const loginuser = await User.findOne({ username: req.body.username })
+        if (!loginuser) {
+            res.redirect('/');
+        } else {
```
Comment (MadaraUchiha) [inline server.js]: The idea in doing it this was is that you can `return` inside the if, then you don't need the `else`, keeping you code flat

File: server.js [hunk 2]
```diff
@@ -23,47 +23,43 @@ app.use(cookieParser());
...
-        const curentuser = await User.findOne({username: currentSession.user})
-        console.log(curentuser);
-        console.log(curentuser.id);
-        console.log(currentSession.id);
-        if (req.params.id !== curentuser.id) {
-            res.redirect(`/profile/${curentuser.id}`)
+        const tryingUser = await User.findById(req.params.id)
+        if (!tryingUser) {
+            res.redirect('/')
         } else {
```
Comment (MadaraUchiha) [inline server.js]: Similarly here

File: server.js [hunk 3]
```diff
@@ -23,47 +23,43 @@ app.use(cookieParser());
...
-        const currentSession = await Session.findById(cookies.session_id)
-        if (currentSession === null) {
-            res.render('index', currentSession);
+        if (!req.cookies.session) {
+            res.render('index');
         } else {
-            res.render('index', currentSession);
+            const currentUser = await User.findById(req.cookies.session.user);
+            if (currentUser) {
+                res.redirect(`profile/${req.cookies.session.user}`);
```
Comment (MadaraUchiha) [inline server.js]: If a user can only access their own profile, why even add the user ID to the url? You know which user it is by their session

File: server.js [hunk 4]
```diff
@@ -93,34 +88,32 @@ app.post("/signup", async (req, res) => {
...
+            } else {
                 const newSession = await Session.create({
-                    user: req.body.username,
+                    user: loginuser.id,
                 });
-                res.cookie('session_id', newSession.id);
-                console.log(loginuser.id);
-                res.redirect(`/profile/${loginuser.id}`);
-            } else {
+                res.cookie('session', newSession);
```
Comment (MadaraUchiha) [inline server.js]: So you give me, the user, the full session? What's stopping me from editing it and putting whatever user ID I want there?

The idea is that the user only gets the session ID, and only the server knows what kind of data is associated with each session. Else the user can simply change their cookie to say "I am admin" and your code would simply believe that to be true.

**Review by MadaraUchiha — CHANGES_REQUESTED** [review]
Most of it is small, cosmetic stuff. But the thing with the session vs session ID is important.

File: server.js (hunk 4 above)
Comment (Fiamma-Jikukan) [reply server.js]: I agree with everything else (staga dush) but the thing that's stopping you from editig it is the database. Only exisiting users can reach the profile page

Comment (MadaraUchiha) [reply server.js]: Cool, so I can just change the userID in my session object and I can be any other user, no need for their password :D

Comment (Fiamma-Jikukan) [reply server.js]: RIP

File: server.js [hunk 5]
```diff
@@ -23,50 +23,48 @@ app.use(cookieParser());
...
-            } else {
-                res.redirect('/')
-            }
+        if (!req.cookies.session) {
+            res.redirect('/')
+            return;
         }
-
+        const currentSession = await Session.findById(req.cookies.session);
+        const currentUser = await User.findById(currentSession.user);
```
Comment (MadaraUchiha) [inline server.js:57]: What happens if I do have a session, but it's not in the database? Remember, the client can send you whatever it wants.

Comment (MadaraUchiha) [inline server.js]: What happens if I do have a session, but it's not in the database? Remember, the client can send you whatever it wants.

File: server.js [hunk 6]
```diff
@@ -23,50 +23,48 @@ app.use(cookieParser());
...
+            return;
         }
-
+        console.log(req.cookies.session);
+        const currentSession = await Session.findById(req.cookies.session);
+        if (!currentSession) {
+            res.clearCookie('session');
+            res.render('index');
+            return;
+        }
```
Comment (MadaraUchiha) [inline server.js:52]: Here you did it correctly, you check to see if the session you got is even valid.

File: server.js [hunk 7]
```diff
@@ -77,50 +75,47 @@ app.post("/signup", async (req, res) => {
     try {
```
Comment (MadaraUchiha) [inline server.js:94]: Not strictly related to this change, but why is `await User.findOne(...)` outside of the `try`? If anything might throw, this is it.

**Review by MadaraUchiha — COMMENTED** [review]
Looks good! Just a couple of small things

**Review by MadaraUchiha — APPROVED** [review]

**Pull request merged**